### PR TITLE
MINOR: Improve ListConsumerGroupTest.testListGroupCommand

### PR DESCRIPTION
**Pull request by dajac**
While reviewing https://github.com/apache/kafka/pull/15150, I found that our tests verifying the console output are really hard to read. Here is my proposal to make it better.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by dajac** [comment]
@mimaison @nizhikov @rreddy-22 @jolshan Could you take a look when you get a chance?

**Comment by nizhikov** [comment]
Hello, @dajac I will take a look, shortly.

File: tools/src/test/java/org/apache/kafka/tools/consumer/group/ListConsumerGroupTest.java [hunk 1]
```diff
@@ -123,42 +128,89 @@ public void testListGroupCommand(String quorum) throws Exception {
...
+    }
+
+    /**
+     * Validates that the output of the list command corresponds to the expected values.
+     *
+     * @param args              The arguments for the command line tool.
+     * @param expectedHeader    The expected header as a list of strings; or an empty list
+     *                          if a header is not expected.
+     * @param expectedGroups    The expected groups as a set of list of strings. The list
+     *                          of strings corresponds to the columns.
```
Comment (rreddy-22) [inline tools/src/test/java/org/apache/kafka/tools/consumer/group/ListConsumerGroupTest.java]: can we say expected* columns?

**Review by rreddy-22 — COMMENTED** [review]
Thanks for the PR!

File: tools/src/test/java/org/apache/kafka/tools/consumer/group/ListConsumerGroupTest.java (hunk 1 above)
Comment (jolshan) [reply tools/src/test/java/org/apache/kafka/tools/consumer/group/ListConsumerGroupTest.java]: as an aside, I like that we are being more thorough with checking the components are in the right rows and columns 👍

File: tools/src/test/java/org/apache/kafka/tools/consumer/group/ListConsumerGroupTest.java [hunk 2]
```diff
@@ -123,42 +128,89 @@ public void testListGroupCommand(String quorum) throws Exception {
...
+     * @throws InterruptedException
+     */
+    private static void validateListOutput(
```
Comment (rreddy-22) [inline tools/src/test/java/org/apache/kafka/tools/consumer/group/ListConsumerGroupTest.java:177]: The return type should be boolean no?

**Review by jolshan — APPROVED** [review]

File: tools/src/test/java/org/apache/kafka/tools/consumer/group/ListConsumerGroupTest.java (hunk 2 above)
Comment (jolshan) [reply tools/src/test/java/org/apache/kafka/tools/consumer/group/ListConsumerGroupTest.java:177]: The method itself does the asserts, so I think it is ok to be void.

File: tools/src/test/java/org/apache/kafka/tools/consumer/group/ListConsumerGroupTest.java [hunk 3]
```diff
@@ -123,42 +128,89 @@ public void testListGroupCommand(String quorum) throws Exception {
...
-            return out.get().contains("STATE") && out.get().contains(GROUP) && out.get().contains("Stable");
-        }, "Expected to find " + GROUP + " in state Stable and the header, but found " + out.get());
+            String output = runAndGrabConsoleOutput(args);
+            out.set(output);
+
+            int index = 0;
+            String[] lines = output.split("\n");
+
+            // Parse the header if one is expected.
+            if (expectedHeader.size() > 0) {
```
Comment (rreddy-22) [inline tools/src/test/java/org/apache/kafka/tools/consumer/group/ListConsumerGroupTest.java]: can we use if(!expectedHeader.isEmpty())

File: tools/src/test/java/org/apache/kafka/tools/consumer/group/ListConsumerGroupTest.java [hunk 4]
```diff
@@ -123,42 +128,89 @@ public void testListGroupCommand(String quorum) throws Exception {
...
+                if (lines.length < 1) return false;
```
Comment (rreddy-22) [inline tools/src/test/java/org/apache/kafka/tools/consumer/group/ListConsumerGroupTest.java]: can we use if (lines.length == 0), just for better readability

File: tools/src/test/java/org/apache/kafka/tools/consumer/group/ListConsumerGroupTest.java (hunk 1 above)
Comment (dajac) [reply tools/src/test/java/org/apache/kafka/tools/consumer/group/ListConsumerGroupTest.java]: I used `expectedRows`.

**Review by rreddy-22 — APPROVED** [review]
lgtm thanks!

**Pull request merged**